src/java.base/share/classes/jdk/internal/constant/MethodTypeDescImpl.java

```diff
@@ -105,18 +106,66 @@ public static MethodTypeDescImpl ofValidated(ClassDesc returnType, ClassDesc...
      * @jvms 4.3.3 Method Descriptors
      */
     public static MethodTypeDescImpl ofDescriptor(String descriptor) {
-        // Implicit null-check of descriptor
-        List<ClassDesc> ptypes = ConstantUtils.parseMethodDescriptor(descriptor);
-        int args = ptypes.size() - 1;
-        ClassDesc[] paramTypes = args > 0
-                ? ptypes.subList(1, args + 1).toArray(ConstantUtils.EMPTY_CLASSDESC)
-                : ConstantUtils.EMPTY_CLASSDESC;
-
-        MethodTypeDescImpl result = ofValidated(ptypes.get(0), paramTypes);
+        int rightBracket, len;
+        if (descriptor.charAt(0) != '('
+                || (rightBracket = (descriptor.charAt(1) == ')' ? 1 : descriptor.lastIndexOf(')'))) <= 0
+                || (len = descriptor.length() - rightBracket - 1) == 0
+                || (len != 1 && len != ConstantUtils.skipOverFieldSignature(descriptor, rightBracket + 1, descriptor.length()))
```

Comment:
Has this be tested against input values like `()A`? I would prefer to fail explicitly with `descriptor.charAt(rightBracket + 1) == 'V'` instead of using `len != 1` and relying on `Wrapper.forBasicType`.

Reply:
Yes, if length == 1, it now relies on Wrapper.forBasicType to detect illegal values.

Reply:
We just exposed an internal exception behavior to public. Might add in wrapper.forBasicType that this IAE is part of public API.

Reply:
I've added a return type check for length == 1, and now the error behavior is the same as before.